File: Leetcode/CS/Leetcode665-非递减数列.cs
public class Solution {
    public bool CheckPossibility(int[] nums) {
        int n = nums.Length;
        int cnt = 0;
        for(int i = 0;i < n-1;i++){
            int x = nums[i],y = nums[i+1];
            if(x > y){
                cnt++;
                if (cnt > 1){
                    return false;
                }
                if(i > 0 && nums[i+1] < nums[i - 1]){
                    nums[i+1] = nums[i]; 
                }
            }
        }
        return true;
    }
}

File: Leetcode/Kotlin/Leetcode665-非递减数列.kt
class Solution {
    fun checkPossibility(nums: IntArray): Boolean {
        val n = nums.size
        var cnt = 0
        for (i in 0 until n-1){
            val x = nums[i]
            val y= nums[i+1]
            if(x > y){
                cnt++
                if(cnt > 1){
                    return false    
                }
                if(i > 0 && nums[i+1] < nums[i-1]){
                    nums[i+1] = nums[i]
                }
            } 
        }
        return true
    }
}